v2/concordium/types.proto

```diff
@@ -589,6 +589,27 @@ message InstanceInfo {
   }
 }
 
+// A smart contract instance key-value pair.
+message InstanceStateKVPair {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+// Request for a specific key of a smart contract instance state.
+message InstanceStateLookupRequest {
+  // The block to be used for the query.
+  BlockHashInput block_hash = 1;
+  // The address of the smart contract instance.
+  ContractAddress address = 2;
+  // Key to look up. If the instance is a V0 instance then this will not be used.
+  bytes key = 3;
```

Comment:
Consider making a separate type for v0 / v1 lookups.

Reply:
I have considered the following options
- only have this supported for V1 contracts. For V0 you get the same information using GetInstanceState, with the same performance. But that means this V1 query would have to fail for V0 contracts, since addresses are the same we cannot really apriori distinguish those, so we have to handle that failure.
- Have this key optional, but that also introduces additional failure modes which maybe are not worth it.

I do share the feeling that this is not "right", but I also don't see that the additional complexiy would make it better.